CoreService/Workflow.psm1

```diff
@@ -0,0 +1,231 @@
+					
+					if($Status -eq $null)
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Any;				
+					}
+					if($Status -eq "Active")
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;				
+					}
+					if($Status -eq "Historical")
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Historical;				
+					}			
+								
+					$WorkflowItemDetail = $client.GetSystemWideList($filter);					
+					return $WorkflowItemDetail
+				}
+			    'AssignedToById'
+				{
+					Write-Verbose "Loading Publication Target with Id '$AssignedToById'..."
+					if ($AssignedToById -ne $null)
+					{
+						if(!$AssignedToById.EndsWith('-65552'))
+						{						
+							if(!$AssignedToById.EndsWith('-65568'))
+							{
+								Write-Error "'$AssignedToById' is not a valid User or Group TCM URI.";
+								return;
+							}
+							else
+							{
+								$groupData = Get-TridionGroup -Id $AssignedToById
+							    if($groupData -eq $null)
+								{
+									Write-Error "'$AssignedToById' is not a valid Group TCM URI.";
+									return;
+								}
+							}
+						}
+						else
+						{
+							$userData = Get-TridionUser -Id $AssignedToById
+							if($userData -eq $null)
+							{
+								Write-Error "'$AssignedToById' is not a valid User TCM URI.";
+								return;
+							}							
+						}					
+					}
+					$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;
+					$WorkflowItemDetail = $client.GetSystemWideList($filter);
+					if ($WorkflowItemDetail -ne $null)
+					{
+						foreach($item in $WorkflowItemDetail)
+						{
+							$processInstanceData = $client.Read($item.Id,$readOption);
+							$acvities = $processInstanceData.Activities | Select-Object -Last 1;
+							$assignee = $acvities.Assignee.IdRef;
+							if ($assignee -eq $AssignedToById)
+							{
+							  $list.Add($processInstanceData)
+							}
+						}
+						return $list;
+					}
+					else
+					{
+						Write-Error "No active workflow item found.";
+						return $null;
+					}					
+				}
+				'AssignedToByTitle'
+				{
+					if ($AssignedToByTitle -ne $null)
+					{
+						if([regex]::Match($AssignedToByTitle,'.*?\\.*?').Success)
+						{
+							$userData = Get-TridionUser -Title $AssignedToByTitle
+							if($userData -eq $null)
+							{
+								Write-Error "'$AssignedToByTitle' is not a valid User Name.";
+								return;
+							}
+						}
+						else
+						{
+							$groupData = Get-TridionGroup -Title $AssignedToByTitle
+							if($groupData -eq $null)
+							{
+								Write-Error "'$AssignedToByTitle' is not a valid Group Name.";
+								return;
+							}
+						}
+						
+					}
+					Write-Verbose "Loading Publication Target with title '$AssignedToByTitle'..."
+					$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;
```

Comment:
Similarly, this cannot be easily tested -- so we should move it to a utility method that can be mocked up.